### DISCUSSION TITLE
all recent nanomq-full images broken - working image with sqlite?

### DISCUSSION THREAD

**Discussion by georgfaust** [nanomq/nanomq#1564] [Q&A] [answered]
With the 18.2-full image mentioned in the docs, sqlite rules do not seem to do anything. Is this a known problem or should that work?

I also tried later versions, they all exit with 
```
 nanomq: error while loading shared libraries: libmysqlclient.so.18: cannot open shared object file: No such file or directory
```

I'm on a M2 Macbook.

**Comment by JaylinYu** [comment]
> With the 18.2-full image mentioned in the docs, sqlite rules do not seem to do anything. Is this a known problem or should that work?
> 
> 
> 
> I also tried later versions, they all exit with 
> 
> ```
> 
>  nanomq: error while loading shared libraries: libmysqlclient.so.18: cannot open shared object file: No such file or directory
> 
> ```
> 
> 
> 
> I'm on a M2 Macbook.

Sorry, bro. We have Nothing mentioned about MySQL integration in Docs. No docker release image has MySQL embedded. Not even in binary release. You have to compile yourself. 
As for cross-compiling of this feature Plz find more information here https://github.com/nanomq/nanomq/issues/1295

**Comment by georgfaust** [comment]
I dont even want MySQL (all version later that 0.18-full I tried just directly exit with this msg).

I just want to evaluate rules that write stuff to sqlite.

Can you confirm, that there is a image that works for sqlite?
I'm quite sure I got a correct config:

```
sqlite {
	disk_cache_size = 102400
	flush_mem_threshold = 100
	resend_interval = 5000
}

rules.sqlite {
	path = "/tmp/sqlite_rule.db"
	rules = [
		{
			sql = "SELECT topic, payload FROM \"abc\""
			table = broker1
		}
	]
}
```

but publishing anything to topic "abc" does not result in an entry in sqlite with 0.18-full.
I do not get anything in the logs, not even an error.
Is there any way to enable monitoring of rules like in EMQX?

**Comment by JaylinYu** [comment]
The log you posted really confuses me. It indicates you are using MySQL plugin.

SQLite should already haven been supported in full image (after 0.19.5) by default, if not, that's a bug. will look into it.

Could you confirm the release version in which you experienced issues? also with the output logs.
because that error log above has nothing to do with sqlite

**Comment by ahsf** [comment]
So make things more clear, there are two different problems:

1. First I tried sqlite following the docs with 0.18.2. It did not work with the config posted above.
2. I then tried latest full which is 0.20-full I think, also tried 0.20.6-full with those I get the error `nanomq: error while loading shared libraries: libmysqlclient.so.18: cannot open shared object file: No such file or directory` in the **docker** logs. Container exits directly after start with this log message.

Here are the images I tested with.

https://hub.docker.com/layers/emqx/nanomq/0.20.6-full/images/sha256-463c72c952c443b67c758e52fdd318edb118221aca7510a28120550e73f39241

https://hub.docker.com/layers/emqx/nanomq/0.20-full/images/sha256-e0d4fdb0afe6c7750fff9dd3b88622b08fe221cde1f77f3a1f41986c75a5f635

libmysql is installed here: https://github.com/nanomq/nanomq/blob/243d8af6f7c8b87195c0188fdd6ea4ea4f10df09/deploy/docker/Dockerfile-full#L7

linked here:
https://github.com/nanomq/nanomq/blob/243d8af6f7c8b87195c0188fdd6ea4ea4f10df09/nanomq/CMakeLists.txt#L77

maybe this helps (inside 0.20-full, nothing changed just `apt-get install mlocate`)
````
root@236def24515e:/etc# nanomq
nanomq: error while loading shared libraries: libmysqlclient.so.18: cannot open shared object file: No such file or directory
root@236def24515e:/etc# locate libmysqlclient.so.18
root@236def24515e:/etc# locate libmysqlclient.so   
/usr/lib/aarch64-linux-gnu/libmysqlclient.so
/usr/lib/aarch64-linux-gnu/libmysqlclient.so.21
/usr/lib/aarch64-linux-gnu/libmysqlclient.so.21.2.35
```

**Comment by JaylinYu** [comment, marked as answer]
Thanks for reporting. It seems like we accidentally introduced a bug when adding SQLite as a default dependency in the release binaries. This brings libmysqlclient as a default compilation when enabling rule_engine.

Now it has been fixed in https://github.com/nanomq/nanomq/pull/1567, libmysqlclient will be excluded.

thank you again for identifying bugs!
